### PR TITLE
Replace imgui with egui in shells

### PR DESCRIPTION
**Pull request by dragonnn**
This PR replaces imgui with egui. All shell elements have bean already rewritten to use egui.
Only one know issue is left: the first texture after egui is draw is not displayed on the screen, if we add for example a dummy draw of a texture with 0,0 size it starts working so I suspect a gl command missing in smithay-egui draw function?

### CODE REVIEW

**Comment by dragonnn** [comment]
Hmmm.... I have everything merged, why it is is falling. Is that again xwayland?
cargo check --all-features seams to enable it :|

**Comment by dragonnn** [comment]
Found a hacky workaround around the issue with the first window not drawing, needs a better solution but will work for now

**Comment by PolyMeilex** [comment]
Hi! Don't worry anout rendering problems, after rebase to my branch it will no longer use this method of rendering anyway.

File: Cargo.toml [hunk 1]
```diff
@@ -64,6 +65,11 @@ features = ["composite"]
 version = "0.9.0"
 features = ["executor", "futures-io"]
 
+[dependencies.smithay-egui]
+git = "https://github.com/dragonnn/smithay-egui"
+commit = "1a8a03304eb8fdd5d770144de59f4ea9c0b94d56"
```
Comment (PolyMeilex) [inline Cargo.toml]: Isn't this field called `rev`? Cargo complains that this key does not exist.
[Cargo Book Specifying dependencies from git repositories](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories)

Comment (dragonnn) [reply Cargo.toml]: yes, you are right. No idea from where I got the commit idea

File: src/output_map/output.rs [hunk 2]
```diff
@@ -130,14 +125,18 @@ impl Output {
...
-        imgui_pipeline: Renderer,
         active_workspace: String,
         logger: slog::Logger,
     ) -> Self
     where
         N: AsRef<str>,
     {
+        let egui = EguiState::new();
+        let mut visuals = egui::style::Visuals::default();
+        visuals.window_corner_radius = 0.0;
```
Comment (PolyMeilex) [inline src/output_map/output.rs]: Minor nitpick, visuals does not have to be mutable in order to construct it with default.
```suggestion
        let visuals = egui::style::Visuals {
               window_corner_radius: 0.0,
               ..Default::default()
        };
```

Comment (dragonnn) [reply src/output_map/output.rs]: Oh, nice, good idea. Changed :)

**Pull request merged**